src/transport/Message.cpp

```diff
@@ -0,0 +1,59 @@
+#include <faabric/transport/Message.h>
+
+#include <faabric/util/logging.h>
+
+namespace faabric::transport {
+Message::Message(const zmq::message_t& msgIn)
+  : _size(msgIn.size())
+  , _more(msgIn.more())
+  , _persist(false)
+{
+    msg = reinterpret_cast<uint8_t*>(malloc(_size * sizeof(uint8_t)));
+    memcpy(msg, msgIn.data(), _size);
+}
+
+Message::Message(int sizeIn)
+  : _size(sizeIn)
+  , _more(false)
+  , _persist(false)
+{
+    msg = reinterpret_cast<uint8_t*>(malloc(_size * sizeof(uint8_t)));
+}
+
+// Empty message signals shutdown
+Message::Message()
+  : msg(nullptr)
+{}
+
+Message::~Message()
+{
+    if (!_persist) {
+        free(reinterpret_cast<void*>(msg));
```

Comment:
Note that if the message goes out of scope, but is marked as persistent, we will have a dangling pointer that we need to free ourselves. While risky, this is necessary for the snapshot server.